packages/root-password-protect/src/plugin/plugin.tsx

```diff
@@ -175,9 +182,26 @@ export async function renderPasswordPage(
   props?: Omit<PasswordPageProps, 'nonce'>
 ) {
   const nonce = generateNonce();
-  const mainHtml = renderToString(<PasswordPage {...props} nonce={nonce} />);
-  const html = `<!doctype html>\n${mainHtml}`;
   res.setHeader('content-type', 'text/html');
+  res.status(401);
   setSecurityHeaders(res, nonce);
-  res.send(html);
+  async function getHtml() {
+    // First, try to render the `/401` route. The route served at this path will
+    // display the "not authorized" interstitial page with a password prompt. The
+    // page should have a form on it that sends a POST request to itself with a
+    // form that contains a single field "password".
+    const route = req.renderer?.getRoute('/401')?.[0];
+    if (route) {
+      const data = await req.renderer?.renderRoute(route, {
+        routeParams: {nonce},
```

Comment:
we should reserve `routeParams` to be exclusively used for the url placeholder params. i'm thinking we should probably add some method to the renderer to handle how this is rendered and have the nonce logic handled by the root renderer.

(we'll probably also want to respect the `route.handle()` method, which is where we can pass any props to the component)

i can help with this more tomorrow when i have more time to think about this

Reply:
yeah; sounds good. also, i realized i can just vendor the plugin into the project with the deadline and customize the page as a one-off, so we don't need to rush this for the time being. however, it'd probably be nice for sites to customize the password entry page so we should fix this up when time permits so sites have a canonical way of customizing it :)